src/databricks/labs/remorph/reconcile/query_builder/tsql_sampling_query.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+import sqlglot.expressions as exp
+from pyspark.sql import DataFrame
+from sqlglot import select
+
+from databricks.labs.remorph.config import get_key_from_dialect
+from databricks.labs.remorph.reconcile.query_builder.base import QueryBuilder
+from databricks.labs.remorph.reconcile.query_builder.expression_generator import (
+    build_column,
+    build_literal,
+    _get_is_string,
+    build_join_clause,
+    trim,
+    coalesce,
+)
+
+_SAMPLE_ROWS = 50
+
+logger = logging.getLogger(__name__)
+
+
+def _union_concat(
+    unions: list[exp.Select],
+    result: exp.Union | exp.Select,
+    cnt=0,
+) -> exp.Select | exp.Union:
+    if len(unions) == 1:
+        return result
+    if cnt == len(unions) - 2:
+        return exp.union(result, unions[cnt + 1])
+    cnt = cnt + 1
+    res = exp.union(result, unions[cnt])
+    return _union_concat(unions, res, cnt)
+
+
+class TsqlSamplingQueryBuilder(QueryBuilder):
```

Comment:
why do we need a dedicated sampler?

Reply:
The current Sampling query building class uses With Cte which fails for Sql Server 
Thus created custom sampler 

Trying a different approach now using prepare query option as mentioned here : https://github.com/microsoft/mssql-jdbc/issues/1340

Reply:
```
Failed -> An error occurred while calling o624.load.
: com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near the keyword 'WITH'.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:265)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1676)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:620)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:540)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7627)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:3916)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:268)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:242)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeQuery(SQLServerPreparedStatement.java:459)
	at org.apache.spark.sql.execution.datasources.jdbc.JDBCRDD$.$anonfun$getQueryOutputSchema$2(JDBCRDD.scala:69)
	at scala.util.Using$.resource(Using.scala:269)
	at org.apache.spark.sql.execution.datasources.jdbc.JDBCRDD$.$anonfun$getQueryOutputSchema$1(JDBCRDD.scala:67)
	at scala.util.Using$.resource(Using.scala:269)
```